polaris-react/src/components/InContextLearning/InContextLearning.tsx

```diff
@@ -29,12 +24,15 @@ export function InContextLearning({children, steps, onDismiss}: Props) {
   const i18n = useI18n();
   const hasMultipleSteps = steps.length > 0;
   const [currentStep, setCurrentStep] = useState(0);
-  const [currentPosition, setCurrentPosition] = useState<Position | null>(null);
-  const wrapperRef = useRef<HTMLInputElement>(null);
-  const wrapperRefHeight = wrapperRef.current?.getBoundingClientRect().height;
+  const [currentActivator, setCurrentActivator] = useState<HTMLElement | null>(
+    null,
+  );
 
   useEffect(() => {
-    setCurrentPosition(updatePosition(steps[currentStep], wrapperRefHeight));
+    setCurrentActivator(
+      (document.querySelector(steps[currentStep].selector) as HTMLElement) ??
```

Comment:
I would love to know how to avoid this cast. `querySelector` returns an `Element` but the `PositionedOverlay.activator` prop expects a `HTMLElement` 🤦

Reply:
I thiiiiink `querySelector` is a generic, so you should be able to do: `document.querySelector<HTMLElement>(steps[currentStep.activator)`

Reply:
Thank you!